test/Microsoft.ML.Tests/Scenarios/Api/EstimatorPipe.cs

```diff
@@ -0,0 +1,337 @@
+﻿using Microsoft.ML.Core.Data;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using Microsoft.ML.Runtime.Model;
+using Microsoft.ML.Tests.Scenarios.Api;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+[assembly: LoadableClass(typeof(ITransformer), typeof(TransformerChain), null, typeof(SignatureLoadModel),
+    "Transformer chain", TransformerChain.LoaderSignature)]
+
+namespace Microsoft.ML.Tests.Scenarios.Api
+{
+    public sealed class TransformerChain<TLastTransformer> : ITransformer, ICanSaveModel
+        where TLastTransformer : class, ITransformer
+    {
+        private readonly ITransformer[] _transformers;
```

Comment:
>[] [](start = 37, length = 2)

any reason why it's array?

Reply:
it's a fixed sequence of objects that have a common type. I think array is a natural choice for such a structure.

---
In reply to: [210998941](https://github.com/dotnet/machinelearning/pull/688#discussion_r210998941) [](ancestors = 210998941)